File: python/프린터큐(1966).py
T = int(input())

for _ in range(T):
    n, m = tuple(map(int, input().split()))
    data = list(map(int, input().split()))

    result = 1
    while data:
        if data[0] < max(data):
            data.append(data.pop(0))
        else:
            if m == 0:
                break
            data.pop()
            result += 1
        m = m-1 if m > 0 else len(data)-1

    print(result)

File: js/프린터큐(1966).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const t = Number(input[0]);

let index = 1;

for (let i=0; i < t; i++) {
    let [n, m] = input[index].split(' ').map(Number);
    const data = input[index+1].split(' ').map(Number);

    index += 2;

    let result = 1;

    while (data.length > 0) {
        if (data[0] < Math.max(...data)) {
            data.push(data[0]);
            data.shift();
        } else {
            if (m === 0) break;
            data.shift();
            result+=1;
        }
        if (m > 0) {
            m = m -1;
        } else {
            m = data.length-1;
        }
    }

    console.log(result);
}